quay_setup.sh

```diff
@@ -1,12 +1,41 @@
 # oc create -f https://raw.githubusercontent.com/minio/minio-operator/master/minio-operator.yaml
+#Create a Project
+oc new-project quay-enterprise
 # install quay operator
 oc apply -f quay-op.yaml
 # install cso operator
 oc apply -f cso.yaml
 # add pull secrete for quay images
 oc create secret generic redhat-pull-secret --from-file=".dockerconfigjson=docker_quay.json" --type='kubernetes.io/dockerconfigjson'
+#Quay-operator takes some time to come(especially on crc) up so need to wait before bringing up quayecosystem
+#sleep 5m
+
+#code to wait for quay-operator to come up
+req="No resources found in quay-enterprise namespace."
+while true; do
+  echo "Waiting for QuayEcosystem...."
+  VAR1=$(oc get QuayEcosystem 2>&1)
+  if [[ $VAR1 == $req ]]
+  then
+    echo "QuayEcosystem found"
+    oc apply -f quayecosystem.yaml
+    break
+  fi
+  sleep 1
+done
```

Comment:
This can be a function, so that it can be reused.

Reply:
Okay.

Reply:
I have created a function which works for the first case.But for checking whether the secret is populated or not I am unable to make the command run ```oc get secrets quay-enterprise-config-secret -o json | jq -r '.data["config.yaml"]'``` jq compile error.

Reply:
Fixed.